voxblox_ros/src/simulation_server.cc

```diff
@@ -0,0 +1,417 @@
+
+  getServerConfigFromRosParam(nh_private);
+
+  tsdf_gt_.reset(new Layer<TsdfVoxel>(voxel_size_, voxels_per_side_));
+  esdf_gt_.reset(new Layer<EsdfVoxel>(voxel_size_, voxels_per_side_));
+
+  tsdf_test_.reset(new Layer<TsdfVoxel>(voxel_size_, voxels_per_side_));
+  esdf_test_.reset(new Layer<EsdfVoxel>(voxel_size_, voxels_per_side_));
+
+  tsdf_integrator_.reset(
+      new MergedTsdfIntegrator(tsdf_integrator_config, tsdf_test_.get()));
+
+  esdf_integrator_.reset(new EsdfIntegrator(
+      esdf_integrator_config, tsdf_test_.get(), esdf_test_.get()));
+
+  if (generate_occupancy_) {
+    occ_test_.reset(new Layer<OccupancyVoxel>(voxel_size_, voxels_per_side_));
+
+    OccupancyIntegrator::Config occ_integrator_config;
+    occ_integrator_config.max_ray_length_m =
+        tsdf_integrator_config.max_ray_length_m;
+    occ_integrator_.reset(
+        new OccupancyIntegrator(occ_integrator_config, occ_test_.get()));
+
+    EsdfOccIntegrator::Config esdf_occ_config;
+    esdf_occ_config.max_distance_m = esdf_max_distance_;
+    esdf_occ_config.default_distance_m = esdf_max_distance_;
+    esdf_occ_integrator_.reset(new EsdfOccIntegrator(
+        esdf_occ_config, occ_test_.get(), esdf_test_.get()));
+  }
+
+  // ROS stuff.
+  // GT
+  tsdf_gt_pub_ = nh_private_.advertise<pcl::PointCloud<pcl::PointXYZI> >(
+      "tsdf_gt", 1, true);
+  esdf_gt_pub_ = nh_private_.advertise<pcl::PointCloud<pcl::PointXYZI> >(
+      "esdf_gt", 1, true);
+  tsdf_gt_mesh_pub_ = nh_private_.advertise<visualization_msgs::MarkerArray>(
+      "tsdf_gt_mesh", 1, true);
+
+  // Test
+  tsdf_test_pub_ = nh_private_.advertise<pcl::PointCloud<pcl::PointXYZI> >(
+      "tsdf_test", 1, true);
+  esdf_test_pub_ = nh_private_.advertise<pcl::PointCloud<pcl::PointXYZI> >(
+      "esdf_test", 1, true);
+  tsdf_test_mesh_pub_ = nh_private_.advertise<visualization_msgs::MarkerArray>(
+      "tsdf_test_mesh", 1, true);
+
+  view_ptcloud_pub_ = nh_private_.advertise<pcl::PointCloud<pcl::PointXYZRGB> >(
+      "view_ptcloud_pub", 1, true);
+
+  // Set random seed to a fixed value.
+  srand(0);
+}
+
+SimulationServer::SimulationServer(const ros::NodeHandle& nh,
+                                   const ros::NodeHandle& nh_private)
+    : SimulationServer(nh, nh_private, getEsdfMapConfigFromRosParam(nh_private),
+                       getEsdfIntegratorConfigFromRosParam(nh_private),
+                       getTsdfMapConfigFromRosParam(nh_private),
+                       getTsdfIntegratorConfigFromRosParam(nh_private)) {}
+
+bool SimulationServer::generatePlausibleViewpoint(FloatingPoint min_distance,
+                                                  Point* ray_origin,
+                                                  Point* ray_direction) const {
+  // Generate a viewpoint at least min_distance from any objects (if you want
+  // just outside an object, just call this with min_distance = 0).
+  constexpr int max_tries = 50;
```

Comment:
could we expose this or at least move the 50 out of being hidden in the function?